diagnostics/tropical_pacific_sea_level/tropical_pacific_sea_level.py

```diff
@@ -0,0 +1,728 @@
+                    ds_obs = xr.concat([ds_obs,ds_obs_sub],dim='time',data_vars='minimal')
+
+        ############## CMEMS ##############
+        if obs in ['CMEMS']:
+            syear_obs = obs_year_range[nobs][0]
+            fyear_obs = obs_year_range[nobs][1]
+            fmon_obs = obs_year_range[nobs][2]
+            #### create time axis for overlapping period
+            timeax = xr.cftime_range(start=cftime.datetime(syear_obs,1,1),
+                                     end=cftime.datetime(fyear_obs,fmon_obs,1),
+                                     freq='MS')
+            timeax = timeax.to_datetimeindex()    # cftime => datetime64
+            ds_obs['time'] = timeax
+
+            # calculate global mean sea level
+            da_area = sa.da_area(ds_obs, lonname='longitude', latname='latitude',
+                                 xname='longitude', yname='latitude', model=None)
+            da_glo_mean = (ds_obs*da_area)\
+                                .sum(dim=['longitude','latitude'])/\
+                           da_area\
+                                .sum(dim=['longitude','latitude'])
+            ds_obs = ds_obs-da_glo_mean
+
+            # rename
+            ds_obs = ds_obs.rename({'longitude':'lon','latitude':'lat'})
+
+        else:
+            syear_obs = obs_year_range[nobs][0]
+            fyear_obs = obs_year_range[nobs][1]
+            #### create time axis for overlapping period
+            timeax = xr.cftime_range(start=cftime.datetime(syear_obs,1,1),
+                                     end=cftime.datetime(fyear_obs,12,31),
+                                     freq='MS')
+            timeax = timeax.to_datetimeindex()    # cftime => datetime64
+            ds_obs['time'] = timeax
+
+
+        # crop data (time)
+        ds_obs = ds_obs[var]\
+                          .where((ds_obs['time.year'] >= syear)&
+                                 (ds_obs['time.year'] <= fyear)
+                                 ,drop=True)
+        ds_obs = ds_obs\
+                          .where((ds_obs.lat >= np.min(np.array(tp_lat_region)))&
+                                 (ds_obs.lat <= np.max(np.array(tp_lat_region)))
+                                 ,drop=True)
+
+        # store all model data
+        ds_obs_list[var] = ds_obs
+
+        # calculate mean
+        obs_mean_list[var] = ds_obs_list[var].mean(dim='time').compute()
+        ds_obs_list[var] = ds_obs_list[var]-obs_mean_list[var]
+
+        # calculate seasonality
+        obs_season_list[var] = ds_obs_list[var].groupby('time.month').mean(dim='time').compute()
+        ds_obs_list[var] = ds_obs_list[var].groupby('time.month')-obs_season_list[var]
+
+        # remove linear trend
+        obs_linear_list[var] = da_linregress(ds_obs_list[var],stTconfint=0.99)
+
+    obs_linear_mlist[obs] = obs_linear_list
+    obs_mean_mlist[obs] = obs_mean_list
+    obs_season_mlist[obs] = obs_season_list
+    ds_obs_mlist[obs] = ds_obs_list
+
+
+# # detrend
```

Comment:
Leave detailed comments describing why the commented blocks are retained; remove otherwise

Reply:
Agreed. Will remove the commented part.